neurolib/optimize/evolution/evolution.py

```diff
@@ -360,14 +399,16 @@ def evalPopulationUsingPypet(self, traj, toolbox, pop, gIdx):
             # store outputs of simulations in population
             pop[idx].fitness.values = fitnessesResult
             # mean fitness value
-            pop[idx].fitness.score = np.nansum(pop[idx].fitness.wvalues) / (len(pop[idx].fitness.wvalues))
+            #pop[idx].fitness.score = np.nansum(pop[idx].fitness.wvalues) / (len(pop[idx].fitness.wvalues))
+            # works with nans and infs:
+            pop[idx].fitness.score = np.ma.masked_invalid(pop[idx].fitness.wvalues).sum() / (len(pop[idx].fitness.wvalues))
         return pop
 
     def getValidPopulation(self, pop):
-        return [p for p in pop if not np.any(np.isnan(p.fitness.values))]
+        return [p for p in pop if not (np.isnan(p.fitness.values).any() or np.isinf(p.fitness.values).any()) ]
```

Comment:
`np.inf` is now also an invalid result (sometimes preferable to `np.nan` in case you have Multi Objective Optimization, you might have one invalid result as `+np.inf` and the other as `-np.inf`, this doesn't work with `np.nan`)